src/utils/managing.officer.corporate.ts

```diff
@@ -0,0 +1,140 @@
+
+export const getManagingOfficerCorporateById = (req: Request, res: Response, next: NextFunction, backLinkUrl: string, templateName: string) => {
+  try {
+    logger.debugRequest(req, `${req.method} BY ID ${req.route.path}`);
+
+    const id = req.params[ID];
+    const data = getFromApplicationData(req, ManagingOfficerCorporateKey, id, true);
+
+    const principalAddress = (data) ? mapDataObjectToFields(data[PrincipalAddressKey], PrincipalAddressKeys, AddressKeys) : {};
+    const serviceAddress = (data) ? mapDataObjectToFields(data[ServiceAddressKey], ServiceAddressKeys, AddressKeys) : {};
+
+    return res.render(templateName, {
+      backLinkUrl: backLinkUrl,
+      templateName: `${templateName}/${id}`,
+      id,
+      ...data,
+      ...principalAddress,
+      ...serviceAddress
+    });
+  } catch (error) {
+    logger.errorRequest(req, error);
+    next(error);
+  }
+};
+
+export const postManagingOfficerCorporate = async(req: Request, res: Response, next: NextFunction, nextPage: string, registrationFlag: boolean): Promise<void> => {
+  try {
+    logger.debugRequest(req, `${req.method} ${req.route.path}`);
+
+    const data: ApplicationDataType = officerData(req.body, uuidv4());
+
+    const session = req.session as Session;
+    setApplicationData(session, data, ManagingOfficerCorporateKey);
+    if (registrationFlag) {
+      await saveAndContinue(req, session);
+    }
+
+    return res.redirect(nextPage);
+  } catch (error) {
+    logger.errorRequest(req, error);
+    next(error);
+  }
+};
+
+export const updateManagingOfficerCorporate = async(req: Request, res: Response, next: NextFunction, nextPage: string, registrationFlag: boolean): Promise<void> => {
+  try {
+    logger.debugRequest(req, `UPDATE ${req.route.path}`);
+
+    // Remove old Managing Officer
+    removeFromApplicationData(req, ManagingOfficerCorporateKey, req.params[ID]);
+
+    // Set officer data
+    const data: ApplicationDataType = officerData(req.body, req.params[ID]);
+
+    // Save new Managing Officer
+    const session = req.session as Session;
+    setApplicationData(session, data, ManagingOfficerCorporateKey);
+
+    if (registrationFlag) {
+      await saveAndContinue(req, session);
+    }
+
+    return res.redirect(nextPage);
+  } catch (error) {
+    logger.errorRequest(req, error);
+    next(error);
+  }
+};
+
+export const removeManagingOfficerCorporate = async(req: Request, res: Response, next: NextFunction, nextPage: string, registrationFlag: boolean): Promise<void> => {
+  try {
+    logger.debugRequest(req, `REMOVE ${req.route.path}`);
+
+    removeFromApplicationData(req, ManagingOfficerCorporateKey, req.params[ID]);
+    const session = req.session as Session;
+
+    if (registrationFlag) {
+      await saveAndContinue(req, session);
+    }
+
+    return res.redirect(nextPage);
+  } catch (error) {
+    logger.errorRequest(req, error);
+    next(error);
+  }
+};
+
+const officerData = (reqBody: any, id: string): ApplicationDataType => {
```

Comment:
setOfficerData?

Reply:
I believe in [Stephanies PR](https://github.com/companieshouse/overseas-entities-web/pull/684), its set to officerData so I changed it to be the same for consistency.

Reply:
Looks like it's `setOfficerData` there.